homeassistant/components/overkiz/entity.py

```diff
@@ -34,8 +34,33 @@ def __init__(
         self._attr_available = self.device.available
         self._attr_unique_id = self.device.device_url
 
+        if self.is_sub_device and not self.is_sub_device_default_naming:
+            # Check if the provided label is "MAIN_DEVICE-XXX"
+            # In that case, don't use the provided label and use the standard name for each sub devices
+            self._attr_name = self.device.label
+
         self._attr_device_info = self.generate_device_info()
 
+    @property
+    def is_sub_device(self) -> bool:
+        """Return True if device is a sub device."""
+        return "#" in self.device_url and not self.device_url.endswith("#1")
+
+    @property
+    def is_sub_device_default_naming(self) -> bool:
+        """Return True if device has a default name not updated by the user."""
+        if self.is_sub_device and self.device.label and "-" in self.device.label:
+            main_device = self.executor.linked_device(1)
+            if (
+                main_device
+                and main_device.label
+                and self.device.label.startswith(main_device.label + "-")
+                and self.device.label.split("-", 1)[1].isdigit()
+            ):
+                return True
+
+        return False
```

Comment:
You can use a regular expression instead:

```
>>> import re
>>>
>>> #Check if the string starts with "The" and ends with "Spain":
>>>
>>> main_label = "Atlantic"
>>> label = "Atlantic-5"
>>> x = re.search(f"^{main_label}-\d+", label)
>>>
>>> if x:
...   print("YES! We have a match!")
... else:
...   print("No match")
...
YES! We have a match!
```

Reply:
This detection have been removed in the last commit.